workflow/energy_rating_index.rb

```diff
@@ -211,13 +232,23 @@ def read_output(design, designdir, output_hpxml_path)
     vars = "'" + get_all_var_keys(OutputVars.WaterHeatingFuel).join("','") + "'"
     query = "SELECT SUM(ABS(VariableValue)/1000000000) FROM ReportVariableData WHERE ReportVariableDataDictionaryIndex IN (SELECT ReportVariableDataDictionaryIndex FROM ReportVariableDataDictionary WHERE VariableType='Sum' AND KeyValue IN (#{keys}) AND VariableName IN (#{vars}) AND ReportingFrequency='Run Period' AND VariableUnits='J')"
     design_output[:fuelHotWaterBySystem][sys_id] = get_sql_query_result(sqlFile, query)
+
     # Reference Load
     if [Constants.CalcTypeERIReferenceHome, Constants.CalcTypeERIIndexAdjustmentReferenceHome].include? design
       # Only ever conventional storage tank water heater
       vars = "'" + get_all_var_keys(OutputVars.WaterHeatingLoad).join("','") + "'"
       query = "SELECT SUM(ABS(VariableValue)/1000000000) FROM ReportVariableData WHERE ReportVariableDataDictionaryIndex IN (SELECT ReportVariableDataDictionaryIndex FROM ReportVariableDataDictionary WHERE VariableType='Sum' AND KeyValue IN (#{keys}) AND VariableName IN (#{vars}) AND ReportingFrequency='Run Period' AND VariableUnits='J')"
       design_output[:loadHotWaterBySystem][sys_id] = get_sql_query_result(sqlFile, query)
     end
+    # Indirect water system
+    if not design_output[:indirectRelatedHVACBySystem][sys_id].nil?
+      vars = "'" + get_all_var_keys(OutputVars.WaterHeatingHeatExchanger).join("','") + "'"
+      query = "SELECT SUM(ABS(VariableValue)/1000000000) FROM ReportVariableData WHERE ReportVariableDataDictionaryIndex IN (SELECT ReportVariableDataDictionaryIndex FROM ReportVariableDataDictionary WHERE VariableType='Sum' AND KeyValue IN (#{keys}) AND VariableName IN (#{vars}) AND ReportingFrequency='Run Period' AND VariableUnits='J')"
+      design_output[:indirectEnergyHotWaterBySystem][sys_id] = get_sql_query_result(sqlFile, query)
+      vars = "'" + get_all_var_keys(OutputVars.WaterHeatingIndirectBoiler).join("','") + "'"
+      query = "SELECT SUM(ABS(VariableValue)/1000000000) FROM ReportVariableData WHERE ReportVariableDataDictionaryIndex IN (SELECT ReportVariableDataDictionaryIndex FROM ReportVariableDataDictionary WHERE VariableType='Sum' AND  KeyValue IN (#{keys}) AND VariableName IN (#{vars}) AND ReportingFrequency='Run Period' AND VariableUnits='J')"
+      design_output[:indirectBoilerLoadBySystem][sys_id] = get_sql_query_result(sqlFile, query)
+    end
```

Comment:
It seems like you are storing raw combi boiler outputs here and then the combi boiler logic about using these inputs in the ERI calculation are in the `calculate_eri` method. I'm sure that works fine, but I think it would be cleaner to use all of the outputs here to calculate the appropriate `design_output[:fuelHotWaterBySystem]`, `design_output[:fuelHeatingBySystem]`, etc. values, and then the `calculate_eri` method doesn't have to know if there is a combi boiler system or not (and can be left unmodified). If my explanation here is not clear, lets discuss in person.

Reply:
Got what you're saying. I thought this function might be more like reading and storing raw data and more manipulation would happen in that function (so that the manipulation might be easier with handy stored data for all designs). But we only deal with rated home design for combi systems, it would be fine to move it ahead and eliminate the look up. 

Now I switched the logic to make it all done in ``read_output`` Water Heating System block. It ran the same results. And I did find one extra benefit: the output ``ERIRatedHome.csv`` would also reflect the energy splitting which is not achievable in previous approach ( because I didn't touch these design_output hashes). 👍

Reply:
In my mind, the `calculate_eri` method was meant to be building configuration agnostic, it's just the ERI calculations given a bunch of energy/load/etc values. And the energy/load/etc values are obtained in `read_output` (which, admittedly, does more than just read outputs these days). But yes, I can see why you went the path that you did. Glad you're getting the same results both ways!

And I hadn't even thought about ERIRatedHome.csv values when I made my suggestion -- that's definitely a nice benefit of this approach. Thanks for making the change!